packages/hop-node/src/watchers/classes/BaseWatcher.ts

```diff
@@ -181,6 +187,62 @@ class BaseWatcher extends EventEmitter implements IBaseWatcher {
     return `${this.tag}:${key}`
   }
 
+  handleStateSwitch = async () => {
+    if (!this.stateUpdateAddress) {
+      return
+    }
+
+    let state: number
+    try {
+      const l1ChainId = this.chainSlugToId(Chain.Ethereum)
+      const l1Bridge = this.getSiblingWatcherByChainId(l1ChainId).bridge
+      state = await l1Bridge.getStateUpdateStatus(this.stateUpdateAddress, this.bridge.chainId)
+    } catch (err) {
+      this.logger.log(`getStateUpdateStatus failed with ${err}`)
+      return
+    }
+
+    switch (state) {
```

Comment:
This case statement section can replaced and simplified with an enum check


```js
enum State {
  Normal = 0,
  DryMode = 1,
  PauseMode = 2,
  Exit = 3
}

const state: State = Numer(await l1Bridge.getStateUpdateStatus(this.stateUpdateAddress, this.bridge.chainId))
this.setDryMode(state === State.DryMode)
this.setPauseMode(state === State.PauseMode)

if (state === State.Exit) {
  this.logger.error('exit mode enabled')
  this.quit()
}
```

Reply:
Nice call. Resolved with 979427f8a53874ecbadc5887ab87cf17fc9699b6